system/lib/wasmfs/js_api.cpp

```diff
@@ -22,11 +23,14 @@ using namespace wasmfs;
 
 extern "C" {
 
-__wasi_fd_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend);
+// __wasi_fd_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend);
+// int wasmfs_create_directory(char* path, int mode, backend_t backend);
+// int wasmfs_unmount(int dirfd, intptr_t path);
```

Comment:
Is the choice between including emscripten/wasmfs.h and dealing with conflicting backend_t typedefs or repeating the function declarations here? Maybe it would be better to repeate the declarations here and add a TODO about resolving the conflicting typedef and including emscripten/wasmfs.h instead. Sorry for the back-and-forth here.

Reply:
Yeah, I think trying to just wrap the `typedef struct Backend* backend_t` was causing some issues elsewhere where the functions themselves couldn't access `backend_t` without using `wasmfs::backend_t`, so I ended up trying to wrap the whole thing in `namespace wasmfs { ... }`, which caused a few other concerns in this PR. Going with forward declarations might be the simplest approach for this PR and I can try to go back and play with the namespacing in the future.